### PR TITLE
Replace "Master/Slave" with "Manager/Worker"

### PR DESCRIPTION
**Pull request by benas**
This PR updates the code that goes with documentation updates in #717 .

Resolves [BATCH-2834](https://jira.spring.io/browse/BATCH-2834)

### CODE REVIEW

**Comment by benas** [comment]
I just want to mention that XSD changes (see [remote-chunking-master](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-integration/src/main/resources/org/springframework/batch/integration/config/xml/spring-batch-integration-3.1.xsd#L135) and [remote-chunking-slave](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-integration/src/main/resources/org/springframework/batch/integration/config/xml/spring-batch-integration-3.1.xsd#L183) elements) are not included in this PR but they probably should.

**Comment by benas** [comment]
> I just want to mention that XSD changes (see remote-chunking-master and remote-chunking-slave elements) are not included in this PR but they probably should.

This PR has been updated to include XSD changes.

**Comment by PascalSchumacher** [comment]
This pull request seems to contain unrelated changes e.g. `RemoteChunkingManagerStepBuilder.java` etc.

**Comment by benas** [comment]
@PascalSchumacher `RemoteChunkingManagerStepBuilder` is a copy of `RemoteChunkingMasterStepBuilder` (which is deprecated). We did not rename existing classes directly as this would be a breaking change that we don't want to introduce in a minor version. These two classes (and similar ones) will co-exist for some time until we remove the deprecated one.

**Comment by benas** [comment]
> A quick search through the code base lead me to a large number of other places where `master`/`slave` terminology are used. I think we need to make another pass to make sure we have all of the places this is used (I searched for the word `slave` to do my verification).

I guess you are talking about changes in #717 (as mentioned in my first comment). Is that correct?

File: spring-batch-integration/src/main/java/org/springframework/batch/integration/config/annotation/EnableBatchIntegration.java [hunk 1]
```diff
@@ -58,14 +58,14 @@
...
  *
  * 	&#064;Autowired
  * 	private RemoteChunkingWorkerBuilder workerBuilder;
  *
  * 	&#064;Bean
- * 	public TaskletStep masterStep() {
- *       	 return this.masterStepBuilderFactory
+ * 	public TaskletStep managerStep() {
+ *       	 return this.managerStepBuilderFactory
  *       		.get("masterStep")
```
[the remark this answers is not in the document]
Comment (benas) [reply spring-batch-integration/src/main/java/org/springframework/batch/integration/config/annotation/EnableBatchIntegration.java]: good catch. Fixed.

File: spring-batch-integration/src/main/java/org/springframework/batch/integration/partition/RemotePartitioningManagerStepBuilderFactory.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+
+	/**
+	 * Create a new {@link RemotePartitioningManagerStepBuilderFactory}.
+	 * @param jobRepository the job repository to use
+	 * @param jobExplorer the job explorer to use
+	 * @param transactionManager the transaction manager to use
+	 */
+	public RemotePartitioningManagerStepBuilderFactory(JobRepository jobRepository,
+													   JobExplorer jobExplorer, PlatformTransactionManager transactionManager) {
```
[the remark this answers is not in the document]
Comment (benas) [reply spring-batch-integration/src/main/java/org/springframework/batch/integration/partition/RemotePartitioningManagerStepBuilderFactory.java]: fixed

**Comment by mminella** [comment]
You were correct.  Squashed, rebased, and merged as 3743894fa4e287dac0e4752ce65dee15d3afea6d

**Pull request closed without merging**